doc/user_guide/dynamo0p3.rst

```diff
@@ -757,59 +758,73 @@ contained two calls of a kernel with arguments described by the above
-*GH_INTEGER*            *n/a*                         *GH_SUM (Built-ins only)*
-GH_REAL                 n/a                           GH_SUM (Built-ins only)
-GH_FIELD                Discontinuous                 GH_WRITE, GH_READWRITE
-GH_FIELD                Continuous                    GH_INC
-GH_OPERATOR             Any for both 'to' and 'from'  GH_WRITE, GH_READWRITE
-GH_COLUMNWISE_OPERATOR  Any for both 'to' and 'from'  GH_WRITE, GH_READWRITE
-======================  ============================  =========================
-
-.. note:: As mentioned above, note that only Built-ins may modify
-          scalar arguments. In practice this means that the only allowed
-          access for the user-defined scalars in kernels is ``GH_READ``.
-          *Furthermore, since the LFRic infrastructure does not currently
-          support integer reductions, integer scalar arguments are
-          restricted to having read-only access.*
+<dynamo0.3-user-kernel-iterates-over>`). Valid access mode values are
+given in the table below.
+
+.. tabularcolumns:: |l|l|l|
+
++------------------------+------------------------------+--------------------+
+| Argument Type          | Function Space               | Access Type        |
++========================+==============================+====================+
+| GH_INTEGER             | n/a                          | GH_READ            |
++------------------------+------------------------------+--------------------+
+| GH_REAL                | n/a                          | GH_READ            |
++------------------------+------------------------------+--------------------+
+| GH_FIELD               | Discontinuous                | GH_READ, GH_WRITE, |
+|                        |                              | GH_READWRITE       |
++------------------------+------------------------------+--------------------+
+| GH_FIELD               | Continuous                   | GH_READ, GH_INC    |
++------------------------+------------------------------+--------------------+
+| GH_OPERATOR            | Any for both 'to' and 'from' | GH_READ, GH_WRITE, |
+|                        |                              | GH_READWRITE       |
++------------------------+------------------------------+--------------------+
+| GH_COLUMNWISE_OPERATOR | Any for both 'to' and 'from' | GH_READ, GH_WRITE, |
+|                        |                              | GH_READWRITE       |
++------------------------+------------------------------+--------------------+
+
+As mentioned above, only :ref:`Built-ins <dynamo0.3-built-ins>` may
+modify scalar arguments. In practice this means that the only allowed
+access for the scalars in user-defined Kernels is ``GH_READ`` (see the
+allowed accesses for arguments in Built-ins in the
+:ref:`section below <dynamo0.3-built-ins-valid-access>`).
 
 A ``GH_FIELD`` argument that specifies ``GH_WRITE`` or ``GH_READWRITE`` as
```

Comment:
Since we're now only talking about user-defined kernels (which I think is the right thing to do) I think we can remove this bulleted list as we can't have DoFs as an iteration space. Maybe:

"Note also that a ``GH_FIELD`` argument that has ``GH_WRITE``/``GH_READWRITE`` as its access pattern must be on a horizontally-discontinuous function space (see ....). Parallelisation of the loop over the horizontal domain for a kernel that....."

Reply:
Agreed. Rephrased as suggested.